cmd/plugin_test.go

```diff
@@ -311,6 +313,179 @@ func TestStashPluginMain(t *testing.T) {
 	assert.Contains(t, output, amountDownloaded)
 }
 
+// This test addresses the issue mentioned in issue #1645. It creates a directory containing two files.
+// The paths of both files and the directory itself (without any recursive option) are added to the infile,
+// which is then passed to the plugin for upload. The test verifies the following:
+// - Both files are successfully uploaded.
+// - The directory itself is not uploaded, and no empty file with the directory name is created at the destination.
+// - An appropriate message indicating the directory upload failure is returned in the corresponding classad.
+func TestInfileUploadWithDirAndFiles(t *testing.T) {
+
+	server_utils.ResetTestState()
+	t.Cleanup(func() {
+		server_utils.ResetTestState()
+	})
+
+	if os.Getenv("RUN_STASHPLUGIN") == "1" {
```

Comment:
Why is this a check? What happens if RUN_STASHPLUGIN == 0? Is this a leaky environment variable? If that's the case, I think it would be better to find where the issue is and fix the leaked environment variable rather than code against it.

Reply:
The function `stashPluginMain` calls `os.exit` and cannot be tested directly. Therefore, it is executed in its own process. The test itself sets this environment variable while starting itself in a different process, and the variable is set only for the duration of that process.

Below is the link to the snippet where the test runs itself in a separate process with the environment variable set.

https://github.com/PelicanPlatform/pelican/pull/1707/files#diff-7047045de9e6b2f5fc15f5af465c47b95cde0a18745cd7e34ee61faba13267cbR434-R442